packages/gamut-labs/src/experimental/GlobalHeader/AnimatedGlobalHeader/index.tsx

```diff
@@ -0,0 +1,106 @@
+import cx from 'classnames';
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { GlobalHeaderProps } from '..';
+import { BasicGlobalHeader } from '../BasicGlobalHeader';
+import styles from './styles.module.scss';
+
+const defaultScrollingState = {
+  isInHeaderRegion: true,
+  isScrollingDown: true,
+  isScrollingDownFromHeaderRegion: true,
+  prevScrollPosition: typeof window === 'undefined' ? 0 : window?.pageYOffset,
+};
+
+export const AnimatedGlobalHeader: React.FC<GlobalHeaderProps> = (props) => {
+  const [scrollingState, setScrollingState] = useState(defaultScrollingState);
+
+  const {
+    isScrollingDownFromHeaderRegion,
+    isInHeaderRegion,
+    isScrollingDown,
+    prevScrollPosition,
+  } = scrollingState;
+
+  const handleScrolling = useCallback(() => {
+    const currentScrollPosition =
+      typeof window === 'undefined' ? 0 : window?.pageYOffset;
```

Comment:
There's no circumstance in which `window` wouldn't exist but the scroll listener would still fire. You can make this less safe. 😈 

```suggestion
    const currentScrollPosition = window.pageYOffset;
```